src/bosh_azure_cpi/spec/unit/vm_manager/create/availability_set_spec.rb

```diff
@@ -567,25 +577,44 @@
               }
             end
 
-            before do
-              allow(azure_client).to receive(:get_availability_set_by_name)
-                .with(resource_group_name, vm_props.availability_set)
-                .and_return(existing_avset)
+            context 'existing avset is unmanaged' do
+              before do
+                allow(azure_client).to receive(:get_availability_set_by_name)
+                  .with(resource_group_name, vm_props.availability_set)
+                  .and_return(existing_avset)
+              end
+
+              it 'should update the managed property of the availability set' do
+                expect(vm_manager2).to receive(:flock)
+                  .with("#{CPI_LOCK_PREFIX_AVAILABILITY_SET}-#{availability_set_name}", File::LOCK_SH)
+                  .and_call_original
+                expect(vm_manager2).to receive(:flock)
+                  .with("#{CPI_LOCK_PREFIX_AVAILABILITY_SET}-#{availability_set_name}", File::LOCK_EX)
+                  .and_call_original
+
+                expect(azure_client).to receive(:create_availability_set).with(resource_group_name, avset_params)
+                expect(azure_client).to receive(:create_network_interface).twice
+
+                vm_params = vm_manager2.create(instance_id, location, stemcell_id, vm_props, network_configurator, env)
+                expect(vm_params[:name]).to eq(vm_name)
+              end
             end
 
-            it 'should update the managed property of the availability set' do
-              expect(vm_manager2).to receive(:flock)
-                .with("#{CPI_LOCK_PREFIX_AVAILABILITY_SET}-#{availability_set_name}", File::LOCK_SH)
-                .and_call_original
-              expect(vm_manager2).to receive(:flock)
-                .with("#{CPI_LOCK_PREFIX_AVAILABILITY_SET}-#{availability_set_name}", File::LOCK_EX)
-                .and_call_original
+            context 'existing avset is managed' do
```

Comment:
context not clear, suggest: context 'when using managed avset in unmanaged vm', something like that.

Reply:
good.